src/modules/gravityCompensator/main.cpp

```diff
@@ -227,12 +227,31 @@ class gravityModuleCompensator: public RFModule
 
         //------------SPECIAL PARAM TP DEFINE THE HEAD TYPE-----//
         version_tag icub_type;
-        if (rf.check("headV2"))
+        // set icub_type.head_version int to 1 (default) or 2
+        if (rf.check("head_version"))
         {
-            yInfo("'headV2' option found. Using icubV2 head kinematics.\n");
-            icub_type.head_version = 2;
+            if (rf.find("head_version").asDouble()==1.0)
+            {
+                yInfo("\"head_version\" %g requested => Using V1 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+            }
+            else if (rf.find("head_version").asDouble()==2.0 ||
+                rf.find("head_version").asDouble()==2.5)
+            {
+                yInfo("\"head_version\" %g requested => Using V2 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+                icub_type.head_version = 2;
```

Comment:
I'm personally for a `bottom-up` approach, where `bottom` means "we upgrade the remaining modules" and `up` represents the user-interface (i.e. app files and command-line options). 

Otherwise, we risk to hide the fact that e.g. `WBD` is not really doing its new job even though it's equipped with the freshest `--head_version`.

@randaz81 what do you think?